arch/x86_64/mcount-dynamic.c

```diff
@@ -461,18 +461,18 @@ bool add_bad_jump(struct mcount_dynamic_info *mdi, unsigned long callsite,
 static int patch_normal_func(struct mcount_dynamic_info *mdi, struct sym *sym,
 			     struct mcount_disasm_engine *disasm)
 {
-	int instr_size;
 	uint8_t jmp_insn[14] = { 0xff, 0x25, };
 	uint64_t jmp_target;
 	struct mcount_orig_insn *orig;
-	uint64_t sym_addr = sym->addr + mdi->map->start;
+	struct mcount_instrument_info *info;
+
+	info = disasm_check_insns(disasm, mdi, sym);
 
-	instr_size = disasm_check_insns(disasm, mdi, sym);
-	if (instr_size < CALL_INSN_SIZE)
-		return instr_size;
+	if (info->size < CALL_INSN_SIZE)
+		return info->size;
```

Comment:
Who's gonna free the info?

Reply:
last line of patch_normal_func function.